internal/executor/terminalwrapper/terminalwrapper.go

```diff
@@ -71,7 +77,27 @@ func New(
 				subCtx, cancel := context.WithCancel(ctx)
 				defer cancel()
 
-				return wrapper.terminalHost.Run(subCtx)
+				err := wrapper.terminalHost.Run(subCtx)
+				if err != nil {
+					return err
+				}
+
+				if !wrapper.lifecycleStartedSent {
```

Comment:
Can it be moved after the retry and then there will be no need for the field flag. Maybe even put it in a separate retry?

Reply:
I don't think so, because after the retry the terminal is already terminated.

Actually moved it to a locator callback because currently it also waits for the terminal to be terminated and we want it to happen earlier: https://github.com/cirruslabs/cirrus-ci-agent/pull/192/commits/f950824e00226e4873fb0f3e78712c443457f894